### ISSUE TITLE
Reproducing Char 3gram + abbreviation curve

### ISSUE THREAD

**Issue by izuna385** [allenai/scispacy#199]
Hello.
Thanks to your advice(#191), I can reproduce K-recall curve on Medmentions dev data.
Here I set efS=2000 because my interest is linking rather than time performance.
![image](https://user-images.githubusercontent.com/35322641/75129315-c5064280-570b-11ea-8462-901afa895316.png)

So, my another question is how to reproduce char3gram + abb. curve of Figure 3 in the original [paper](https://www.aclweb.org/anthology/W19-5034/).
In paper, unsupervised abbreviation detection and expansion is conducted before ANN search.

My interpretation is, before ANN search, 
https://github.com/allenai/scispacy/blob/master/scispacy/umls_linking.py#L80
is conducted for all mentions, given documents in which mention exists, and then if mention is seen as abbreviation, longer candidate form is used for ANN search. I'm now trying to incorporate this function before ANN search.

In summary, my question is whether this interpretation is correct or not. Also, if there is anything crucial to reproduce this char 3gram + Abb.,  I'd appreciate it if you'd write.

Thanks!

**Comment by DeNeutoy** [comment]
Hi @izuna385 - yes that is correct. The only change is substituting the surface form of a mention for that of it's long form, so `span._.long_form`. To do this, you just need to add the `AbbreviationDetector` in the pipeline before the linker. This is particularly helpful for TF-IDF type candidate generation, because abbreviations are very sparse in terms of 3-grams, meaning their score is not very meaningful unless there is an exact match. Please let me know if you cannot reproduce this, it should be the easier part!

**Comment by DeNeutoy** [comment]
Hi @izuna385 - yes that is correct. The only change is substituting the surface form of a mention for that of it's long form, so `span._.long_form`. To do this, you just need to add the `AbbreviationDetector` in the pipeline before the linker. This is particularly helpful for TF-IDF type candidate generation, because abbreviations are very sparse in terms of 3-grams, meaning their score is not very meaningful unless there is an exact match. Please let me know if you cannot reproduce this, it should be the easier part!

**Comment by izuna385** [comment]
Thanks for reply. Sounds like simple but effective for abbreviation, for `short term(longer term)` and the inverse, which is very often seen at paper. Personally abbreviation expansion is interesting to me, since it's closely related to EL. So in future I may try another algorithm and report results!
Thanks to your advice, I now add pipe before each mention is going to be ANNsearched, and am checking recall per each K. I'll add result when evaluation is complete.

**Comment by izuna385** [comment]
efS=2000, char3gram+abb.

![image](https://user-images.githubusercontent.com/35322641/75342509-ed439c00-58d9-11ea-886f-39d5fc93f4f7.png)

seems like reproduced! Thank you for so much.

**Comment by danielkingai2** [comment]
Seems like we can close this issue, feel free to reopen or make a new one if you have more questions!
